airsonic-main/src/main/java/org/airsonic/player/spring/ServletConfiguration.java

```diff
@@ -4,14 +4,17 @@
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.web.servlet.ViewResolver;
+import org.springframework.web.servlet.config.annotation.ContentNegotiationConfigurer;
+import org.springframework.web.servlet.config.annotation.PathMatchConfigurer;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
 import org.springframework.web.servlet.handler.SimpleUrlHandlerMapping;
 import org.springframework.web.servlet.view.InternalResourceViewResolver;
 import org.springframework.web.servlet.view.JstlView;
 
 import java.util.Properties;
 
 @Configuration
-public class ServletConfiguration {
+public class ServletConfiguration extends WebMvcConfigurerAdapter {
```

Comment:
deprecated class. new mvc doesn't need an adapter, you can use the interface directly.

also you don't need to do this here. you can just enable it in your properties file.

https://github.com/airsonic-advanced/airsonic-advanced/commit/b4f286fd372dd7038801ba1029d59724da64cfe3

Reply:
Good point about the interface.

On the other hand, since we're moving all the configuration to Java, it makes more sense to me to have everything there.

Reply:
we're not moving everything to java. we're moving (have moved?) xml to java.

your application properties remain outside java.
your opinion can differ of course, but consider: this can be autoconfigured by spring boot with one line in the config provided specifically for this reason instead of manually implementing a spring class to override it.

Use the facility provided by the framework.

Reply:
On the other hand Java is type-checked, but the XML/properties files aren't. This is, to me, the main benefit to moving most of the configuration to Java. If the API changes for some reason, then at least the compiler has your back.

Reply:
? I do not understand this comment.

Your bindings will fail if you provide an incorrect config property type. Literally all your tests (that rely on this behavior) will fail. Do you think type checking doesn't happen at binding?

Look, you can continue using this method if you want, my point is that if you're going to use a framework, then you might as well use it and rely on it, including the properties it provides for you. You're not programmatically doing anything exotic to justify maintaining custom class overriding code that now _you_ now have the burden of maintaining, instead of the framework.
The framework gives you specific properties for a particular reason, and I can guarantee they'll do a lot more checking than you will. Switching to Java is justified and valid if the value needs to be dynamically calculated, but none of that is true here.

I mean if you don't want to use the facilities provided by the framework, then you can also write all the wire injection stuff too. It's not hard, but the point is that now _you're_ going to maintain it. I hope my point makes it across.

Reply:
Spring seems to consider properties and code equally, to the point that a lot of documentation mention both at the same time, so it's not like they say "you guys should use properties" in our case.

If the property name changes or the property is removed for some reason, its behavior will silently start to fail. Of course the tests may pick it up, but then they will also fail at some other place with an unrelated error message. Also (in Airsonic we have .properties files in both `src/main` and `src/test`) you may need to maintain additional configuration files for integration tests.

Reply:
Yeah, that's not accurate at all.

If it's documentation you're trying to argue, then here is the actual relevant section of the official Spring Boot documentation for this exact change you're trying to address along with the prescribed solution: https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-spring-mvc-pathmatch

You'll notice that it tells you to use the _properties_, and there is nary a mention of any code.

Leverage the framework you're using; you're not using raw MVC, you're using Boot.

Of course, on this minor matter, you can continue "javafying" if you wish. This point isn't really worth debating over more. I gave my opinion, presented the Spring documentation for it, obviously it's your PR fork to do as you'd like.